components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/TenantCreateGatewayObserver.java

```diff
@@ -0,0 +1,223 @@
+import org.apache.synapse.config.xml.MultiXMLConfigurationSerializer;
+import org.apache.synapse.config.xml.SequenceMediatorFactory;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.apache.synapse.registry.Registry;
+import org.wso2.carbon.appmgt.api.AppManagementException;
+import org.wso2.carbon.appmgt.impl.AppMConstants;
+import org.wso2.carbon.appmgt.impl.utils.AppManagerUtil;
+import org.wso2.carbon.base.CarbonBaseUtils;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.context.RegistryType;
+import org.wso2.carbon.mediation.initializer.configurations.ConfigurationInitilizerException;
+import org.wso2.carbon.mediation.initializer.configurations.ConfigurationManager;
+import org.wso2.carbon.mediation.registry.WSO2Registry;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.utils.AbstractAxis2ConfigurationContextObserver;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+
+/**
+ * This creates the {@link org.apache.synapse.config.SynapseConfiguration}
+ * for the respective tenants. This class specifically add to deploy WebApp Manager
+ * related synapse sequences. This class used to deploy resource mismatch handler, auth failure handler,
+ * sandbox error handler, throttle out handler, build sequence, main sequence and fault sequence into tenant
+ * synapse artifact space.
+ */
+public class TenantCreateGatewayObserver extends AbstractAxis2ConfigurationContextObserver {
+    private static final Log log = LogFactory.getLog(TenantCreateGatewayObserver.class);
+    private String resourceMisMatchSequenceName = "_resource_mismatch_handler_";
+    private String throttleOutSequenceName = "_throttle_out_handler_";
+    private String buildSequenceName = "_build_";
+    private String faultSequenceName = "fault";
+    private String mainSequenceName = "main";
+    private String saml2SequenceName = "saml2_sequence";
+    private String synapseConfigRootPath = CarbonBaseUtils.getCarbonHome() + AppMConstants.SYNAPSE_CONFIG_RESOURCES_PATH;
+    private SequenceMediator authFailureHandlerSequence = null;
+    private SequenceMediator resourceMisMatchSequence = null;
+    private SequenceMediator throttleOutSequence = null;
+    private SequenceMediator sandboxKeyErrorSequence = null;
+    private SequenceMediator productionKeyErrorSequence = null;
+
+
+    public void createdConfigurationContext(ConfigurationContext configurationContext) {
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+    	int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+        try {
+            // first check which configuration should be active
```

Comment:
Formatting issues and Java doc comment issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files